scripts/custom/clean_date.py

```diff
@@ -11,15 +12,22 @@ def clean_date(raw_input):  # noqa: C901
 
     date = None
 
-    # Handle YYYY-MM-DD
+    # Correct format
     try:
-        date = datetime.datetime.strptime(raw_input, "%Y-%m-%d")
-    except ValueError:
+        pattern = re.compile(
+            r"([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)"
+            r"(-(0[1-9]|1[0-2])(-(0[1-9]|[1-2][0-9]|3[0-1])(T([01][0-9]|2[0-3])"
+            r":[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):"
+            r"[0-5][0-9]|14:00)))?)?)?"
+        )
+        full_match = re.fullmatch(raw_input, pattern)
+        date = datetime.datetime.strptime(full_match.group(0)[0:10], "%Y-%m-%d")
```

Comment:
With this, we don't need to handle `YYYY-MM-DDTH:M:S+zzzz` anymore, right?

Reply:
Actually, this handles YYYY-MM-DDTH:M:S+zz:zz (correct format) but not YYYY-MM-DDTH:M:S+zzzz (incorrect)

Reply:
It handles `YYYY-MM-DDTH:M:S+zz:zz`, but not this `YYYY-MM-DDTH:M:S+zzzz` (incorrect)